src/pages/Refund.tsx

```diff
@@ -143,6 +151,21 @@ const Refund = () => {
                         accept="application/json,image/png"
                         onChange={(e) => uploadChange(e)}
                     />
+                    <Show when={swapFound() !== null}>
+                        <hr />
+                        <p>{t("swap_in_progress")}</p>
+                        <button
+                            class="btn btn-success"
+                            onClick={() => navigate(`/swap/${swapFound()}`)}>
+                            {t("open_swap")}
+                        </button>
+                    </Show>
```

Comment:
Always the set `refundJson` and switch between open and refund if `swaps().some((swap) => swap.id === refundJson().id)`